### PR TITLE
Add comments over internal code being used by providers

### LINKED ISSUE #11266: Identify and document internal code used by providers
Following up on the discussion from #11132, we have decided to take the following action for 2.1:
* Identify the internal types that are used by the "big five" providers (Postgres, MySQL, SQLite, SQL Server, SQL Compact) and add a note to each of these types to avoid breaking in the 2.1 timeframe
* File an issue for each class of internal code--e.g. one for resource strings, one for conventions, and so on. For the most part we will address these things post 2.1.
  * If there is already a way to do something without using internals, then file an issue on the relevant providers instead.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #11266

### CODE REVIEW

File: src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ namespace Microsoft.EntityFrameworkCore.Internal
     ///		This API supports the Entity Framework Core infrastructure and is not intended to be used
     ///     directly from your code. This API may change or be removed in future releases.
     /// </summary>
+    // Issue#11266 This method is being used by provider code. Do not break.
```
Comment (smitpatel) [inline src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: While these files are auto-generated. Next generation would remove this warning and diff will point us to make sure not to break.

Comment (AndriySvyryd) [reply src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: Generated files usually aren't reviewed

File: src/EFCore.Relational/Scaffolding/Internal/TableSelectionSet.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace Microsoft.EntityFrameworkCore.Scaffolding.Internal
     ///     directly from your code. This API may change or be removed in future releases.
     /// </summary>
     [Obsolete("This is just to provide backward compatibility for 2.0 providers")]
+    // Issue#11266 This type is being used by provider code. Do not break.
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Scaffolding/Internal/TableSelectionSet.cs]: Redundant comment

File: src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: Alternative?

Comment (AndriySvyryd) [reply src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: Also this is used only in our providers, so it shouldn't be an issue breaking it.

Comment (smitpatel) [reply src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: Other providers are also using Strings files

Comment (AndriySvyryd) [reply src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: You can put comments in the .resx files over the resources that are used

**Pull request merged**